modules/invenio-oaiharvester/invenio_oaiharvester/harvester.py

```diff
@@ -1874,9 +1246,251 @@ def map(self):
+                                    full_key_val_obj = {last_key: value}
+                                    while sub_keys:
+                                        current_key = sub_keys.pop()
+                                        if '[]' in current_key:
+                                            full_key_val_obj = {
+                                                current_key.replace("[]", ""):
+                                                    [full_key_val_obj]}
+                                        else:
+                                            full_key_val_obj = {
+                                                current_key: full_key_val_obj}
+                                    temp_lst.append(
+                                        {"full": full_key_val_obj,
+                                         "key": sub_keys_clone,
+                                         "val": {last_key: value}})
+                            elif 'attributes' in mapping_key.split(".@")[1]:
+                                att = mapping_key.split(".")[-1]
+                                if val_obj.get('@' + att, None):
+                                    att = val_obj['@' + att]
+                                    full_key_val_obj = {last_key: att}
+                                    while sub_keys:
+                                        current_key = sub_keys.pop()
+                                        if '[]' in current_key:
+                                            full_key_val_obj = {
+                                                current_key.replace("[]", ""):
+                                                    [full_key_val_obj]}
+                                        else:
+                                            full_key_val_obj = {
+                                                current_key: full_key_val_obj}
+                                    temp_lst.append(
+                                        {"full": full_key_val_obj,
+                                         "key": sub_keys_clone,
+                                         "val": {last_key: att}})
+                    result_dict = {}
+                    for temp_obj in temp_lst:
+                        merge_dict(result_dict, temp_obj['full'],
+                                   temp_obj['val'], temp_obj['key'])
+                    if isinstance(result_dict[root_key.replace("[]", "")],
+                                  list):
+                        list_result.append(
+                            result_dict[root_key.replace("[]", "")][0])
+                    elif isinstance(result_dict[root_key.replace("[]", "")],
+                                    dict):
+                        list_result.append(
+                            result_dict[root_key.replace("[]", "")])
+            except Exception as ex:
+                import traceback
+                traceback.print_exc()
+            return list_result, root_key.replace("[]", "")
```

Comment:
In case of exception, this line will cause an error because `list_result` and` root_key` have not been initialized (It only initialized in the try statement).